packages/components/src/components/Select/__tests__/Select.spec.tsx

```diff
@@ -0,0 +1,182 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { IconIcon } from '@frontify/fondue-icons';
+import { act, renderHook } from '@testing-library/react';
+import { Children, isValidElement } from 'react';
+import { describe, expect, it, vi } from 'vitest';
+
+import { Select } from '../Select';
+import { type SelectItemProps } from '../SelectItem';
+import { useSelectData } from '../useSelectData';
+import { getSelectOptionValue, isReactLeaf, recursiveMap } from '../utils';
+
+describe('useSelectData', () => {
+    const menuSlot = (
+        <Select.Slot key="test" name="menu">
+            <Select.Item value="test1">Test1</Select.Item>
+            <Select.Item value="test2">Test2</Select.Item>
+            <Select.Group groupId="Group 1">
+                <Select.Item value="test3">Test3</Select.Item>
+                <Select.Item value="test4">Test4</Select.Item>
+                <Select.Item value="test5">Test5</Select.Item>
+            </Select.Group>
+            <Select.Item value="test6">Test6</Select.Item>
+        </Select.Slot>
+    );
+
+    const decoratorsSlot = [
+        <Select.Slot key="test-left" name="left">
+            <IconIcon size={16} />
+        </Select.Slot>,
+        <Select.Slot key="test-right" name="right">
+            <IconIcon size={16} />
+        </Select.Slot>,
+    ];
+    const clearSlot = (
+        <Select.Slot key="test-clear" name="clear">
+            <IconIcon size={16} />
+        </Select.Slot>
+    );
+
+    const selectSlots = [menuSlot, clearSlot, ...decoratorsSlot];
+
+    it('returns menu slots', () => {
+        const {
+            result: { current: result },
+        } = renderHook(() => useSelectData(selectSlots));
```

Comment:
For the hook and utilities, I would use `vitest` (no DOM so wayyy faster)

Reply:
Can you show me an example how you would do it without dom? 
i found it to be the best way to generate react elements that i can pass in.

Because we are comparing the type to the Component inside of the hook, manually constructing the object to mock a react element would be super tedious